lib/internal/process/promises.js

```diff
@@ -58,12 +60,17 @@ function setupPromises(scheduleMicrotasks) {
   }
 
   function emitWarning(uid, reason) {
-    const warning = new Error('Unhandled promise rejection ' +
-                              `(rejection id: ${uid}): ${String(reason)}`);
+    const warning = new Error(
+      `Unhandled promise rejection (rejection id: ${uid}): ` +
+      safeToString(reason));
     warning.name = 'UnhandledPromiseRejectionWarning';
     warning.id = uid;
-    if (reason instanceof Error) {
-      warning.stack = reason.stack;
+    try {
```

Comment:
Do we still need to wrap this block in try-catch? If `reason.stack` throws an error then there must be something really fishy going on..

Reply:
`reason.stack` might not throw an error but the instanceof check might. The [very fishy proxy](https://github.com/nodejs/node/blob/48ef0eead9fd3819bd425f5937581f2df0591d4a/test/parallel/test-promises-unhandled-proxy-rejections.js#L16-L30) I'm using as a test intentionally throws in the [[GetPrototypeOf]] hook to test this condition.